### PR TITLE
[#2701] - Add docker installation instructions and hydroshare deployment

### PR DESCRIPTION
**Pull request by sblack-usu**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. N/A - updated developer documentation

### CODE REVIEW

**Comment by hydrobuild** [comment]
Build finished.

**Comment by aphelionz** [comment]
@sblack-usu Can you prepare a demo for these changes for the Monday 4/30 call?

**Comment by sblack-usu** [comment]
@aphelionz - I'm happy to.

**Comment by hydrobuild** [comment]
Build finished.

File: README.md [hunk 1]
```diff
@@ -18,19 +18,40 @@ This README file is for developers interested in working on the Hydroshare code
...
+5. Set DOCKER_HOST within .bashrc
+    * `echo "export DOCKER_HOST=tcp://0.0.0.0:2375" >> ~/.bashrc && source ~/.bashrc`
+6. Bind /mnt/c to /c (Assuming your drive is c)
+    * `sudo mkdir /c`
+    * `sudo mount --bind /mnt/c /c`
+7. bind mount within .bashrc
+    * `echo "sudo mount --bind /mnt/c /c" >> ~/.bashrc && source ~/.bashrc`
+### Configuration
+1. Clone this repository (the scripts won't work with CLRF line endings so make sure to turn off `core.autocrlf false` 
+2. `cd` into the cloned repo folder
```
Comment (martinseul) [inline README.md]: I needed to cd into the Huydroshare folder and run git config core.autocrlf false. for a newby me it is not clear that this is a git command (Mark pinted this out)

File: README.md [hunk 2]
```diff
@@ -18,19 +18,40 @@ This README file is for developers interested in working on the Hydroshare code
...
+4. [Install Docker-Compose within WSL](https://docs.docker.com/compose/install/)
+5. Set DOCKER_HOST within .bashrc
+    * `echo "export DOCKER_HOST=tcp://0.0.0.0:2375" >> ~/.bashrc && source ~/.bashrc`
+6. Bind /mnt/c to /c (Assuming your drive is c)
+    * `sudo mkdir /c`
+    * `sudo mount --bind /mnt/c /c`
+7. bind mount within .bashrc
+    * `echo "sudo mount --bind /mnt/c /c" >> ~/.bashrc && source ~/.bashrc`
+### Configuration
+1. Clone this repository (the scripts won't work with CLRF line endings so make sure to turn off `core.autocrlf false`
```
Comment (martinseul) [inline README.md]: helpful to link here to simplify : git clone https://github.com/hydroshare/hydroshare

**Comment by sblack-usu** [comment]
So I went on an adventure with docker on windows.  Turns out there is a documented missing feature for mounting volumes from a windows host to a linux docker container.  Basically chown commands won't work in a container on a volume hosted on a windows machine.  https://docs.docker.com/v17.09/docker-for-windows/troubleshoot/#permissions-errors-on-data-directories-for-shared-volumes

By a fluke, I did not run into this problem (and was not aware of it) until I moved my hydroshare folder when trying to organize my dev environment.  When I moved my folder, the chown commands that change /hydroshare from root:root to hydro-service:storage-hydro failed.  After some investigation and testing I discovered that I had created the folder on a shared volume within a linux vm.  Basically, because that folder structure was created on a linux vm, the chown commands worked!

This issue complicates running hydroshare on windows.  I'm happy to document how this workaround works, but I thought I'd let others know about this before updating this PR to hear what they think.  Perhaps @martinseul could try out the workaround.  I plan to continue using Docker for hydroshare as I have been.  I will notify Docker of this workaround, and perhaps they will be able to build this workaround into Docker.

**Comment by aphelionz** [comment]
Document Virtualbox workaround and move from readme

**Pull request closed without merging**